File: src/main/resources/output/IFC4X3_java/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProcess/IfcTaskType/IfcTaskType.java
public class IfcTaskType extends IfcTypeProcess {
    public IfcTaskTypeEnum PredefinedType;
    public IfcLabel WorkMethod;

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : (PredefinedType <> IfcTaskTypeEnum.USERDEFINED) OR ((PredefinedType = IfcTaskTypeEnum.USERDEFINED) AND EXISTS(SELF\IfcTypeProcess.ProcessType))
}

File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProcess/IfcTaskType/IfcTaskType.cs
public class IfcTaskType : IfcTypeProcess
{
    public IfcTaskTypeEnum PredefinedType { get; set; }
    public IfcLabel WorkMethod { get; set; }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : (PredefinedType <> IfcTaskTypeEnum.USERDEFINED) OR ((PredefinedType = IfcTaskTypeEnum.USERDEFINED) AND EXISTS(SELF\IfcTypeProcess.ProcessType))
}
